apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsConstControl.tsx

```diff
@@ -0,0 +1,29 @@
+import type { ControlProps, RankedTester } from "@jsonforms/core"
+import { useEffect } from "react"
+import { and, isStringControl, rankWith, schemaMatches } from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+
+import { JSON_FORMS_RANKING } from "~/constants/formBuilder"
+
+export const jsonFormsConstControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.ConstControl,
+  and(
+    isStringControl,
+    schemaMatches((schema) => schema.const !== undefined),
+  ),
+)
+
+export function JsonFormsConstControl({
+  handleChange,
+  path,
+  schema,
+}: ControlProps) {
+  // TODO: Make the default value persist inside JSONForms
+  useEffect(() => {
+    handleChange(path, schema.const)
+  }, [handleChange, path, schema])
+
+  return null
```

Comment:
this feels abit sus - what it feels like we're trying to do here is to write through to our underlying data structure (our data) but not render anything.

i think the pattern that we might want to aim for here might be to use a [middleware + reducer](https://jsonforms.io/docs/middleware/) so we fire the event only on update and only to mutate data